packages/grafana-ui/src/components/Graph/GraphLegend.tsx

```diff
@@ -11,13 +11,14 @@ import { css } from 'emotion';
 import { selectThemeVariant } from '../../themes/index';
 
 interface GraphLegendProps extends LegendProps {
+  enableSeriesColorChange?: boolean;
```

Comment:
This is named like an action. Maybe canChangeColor or make onSeriesColorChange optional and check if it's provided and use that to enable/disable it.

Reply:
Agree with the naming :thumbsup:

Regarding optional onSeriesColorChange - on this level I want the API to be explicite when it comes to enabling features. Down the hierarchy (GraphLegendListItem) it's handled on onSeriesColorChange being present or not.

Reply:
After second look it will be beneficial to handle optional `onSeriesColorChange` prop only instead of having those two separate properties